cockroachdb/templates/job.init.yaml

```diff
@@ -119,29 +121,140 @@ spec:
           # Command completes either when cluster initialization succeeds,
           # or when cluster has been initialized already.
           command:
-            - /bin/bash
-            - -c
-            - >-
-              while true; do
-              initOUT=$(set -x;
-              /cockroach/cockroach init
-              {{- if .Values.tls.enabled }}
-              --certs-dir=/cockroach-certs/
-              {{- else }}
-              --insecure
-              {{- end }}
-              {{- with index .Values.conf "cluster-name" }}
-              --cluster-name={{.}}
-              {{- end }}
-              --host={{ template "cockroachdb.fullname" . }}-0.{{ template "cockroachdb.fullname" . -}}
-                     :{{ .Values.service.ports.grpc.internal.port | int64 }}
-              2>&1);
-              initRC="$?";
-              echo $initOUT;
-              [[ "$initRC" == "0" ]] && exit 0;
-              [[ "$initOUT" == *"cluster has already been initialized"* ]] && exit 0;
-              sleep 5;
-              done
+          - /bin/bash
+          - -c
+          - >-
+            {{- if $isClusterInitEnabled }}
+              initCluster() {
+                while true; do
+                  local output=$(
+                    set -x;
+
+                    /cockroach/cockroach init \
+                      {{- if .Values.tls.enabled }}
+                      --certs-dir=/cockroach-certs/ \
+                      {{- else }}
+                      --insecure \
+                      {{- end }}
+                      {{- with index .Values.conf "cluster-name" }}
+                      --cluster-name={{.}} \
+                      {{- end }}
+                      --host={{ template "cockroachdb.fullname" . }}-0.{{ template "cockroachdb.fullname" . -}}
+                              :{{ .Values.service.ports.grpc.internal.port | int64 }}
+                  2>&1);
+
+                  local exitCode="$?";
+                  echo $output;
+
+                  if [[ "$exitCode" == "0" || "$output" == *"cluster has already been initialized"* ]]
```

Comment:
@keith-mcclellan is there a way to check if a cluster is up vs grep'ing for a string?

Reply:
I'm not aware of one. This is at least the same as what we were doing before this PR, so I'm fine with continuing to use it.